core/src/main/scala/spinal/core/internals/Phase.scala

```diff
@@ -2476,152 +2477,214 @@ class PhaseCheck_noLatchNoOverride(pc: PhaseContext) extends PhaseCheck{
+                           case Some(branchInfo) =>
+                              if (firstBranchBitsCloned.isEmpty) {
+                                 firstBranchBitsCloned = Some(branchInfo.bits.clone())
+                              } else {
+                                 firstBranchBitsCloned.get.intersect(branchInfo.bits)
+                              }
+                         }
+                      }
+                      intersectionForBt = firstBranchBitsCloned.filter(bits => possibleIntersection && !bits.isEmpty)
+                  }
+
+                  intersectionForBt.foreach { finalBits =>
+                    processAssignment(bt, finalBits, isFullAssignmentCheck = false, location)
                   }
-                }
               }
 
-              for ((bt, assigned) <- mix) {
-                if(getOrEmptyAdd(bt,assigned) && checkOverlap){
-                  PendingError(s"ASSIGNMENT OVERLAP completely the previous one of $bt\n ${s.getScalaLocationLong}")
-                }
+              allTouchedSignals.foreach { bt =>
+                  if (!assigneds.contains(bt)) {
+                      assigneds(bt) = AssignmentInfo(new AssignedBits(bt.getBitsWidth), Some(location))
+                  }
               }
             }
           case s =>
         }
 
 
+        // --- Final Checks (Latch/No-Driver) --- performed only on the second pass (!checkOverlap)
         def finalCheck(bt : BaseType): Unit ={
           // Hold off until suffix parent is processed
           if (bt.isSuffix)
             return
           if (bt.isInstanceOf[Suffixable]) {
-            if (bt.dlcIsEmpty)
-              return bt.asInstanceOf[Suffixable].elements.filter(_._2.isInstanceOf[BaseType]).foreach(e => finalCheck(e._2.asInstanceOf[BaseType]))
+            if (bt.dlcIsEmpty) {
+              bt.asInstanceOf[Suffixable].elements
+                .filter(_._2.isInstanceOf[BaseType])
+                .foreach(e => finalCheck(e._2.asInstanceOf[BaseType]))
+              return
+            }
           }
 
-          val assignedBits = getOrEmpty(bt)
+          val assignmentInfo = assigneds.getOrElse(bt, AssignmentInfo(new AssignedBits(bt.getBitsWidth), None))
+          val assignedBits = assignmentInfo.bits
+
           if ((bt.isVital || !bt.dlcIsEmpty) && bt.rootScopeStatement == body && !assignedBits.isFull){
             if(bt.isComb) {
               val unassignedBits = new AssignedBits(bt.getBitsWidth)
-
               unassignedBits.add(bt.getBitsWidth - 1, 0)
               unassignedBits.remove(assignedBits)
 
               if (!unassignedBits.isEmpty) {
                 if (bt.dlcIsEmpty) {
                   if(!bt.hasTag(allowFloating)) {
-                    PendingError(s"NO DRIVER ON $bt, defined at\n${bt.getScalaLocationLong}")
+                    PendingError(s"NO DRIVER ON $bt (combinatorial signal with no logic connection), defined at\n${bt.getScalaLocationLong}")
```

Comment:
Hmm maybe this could be "(Combinatorial signal with no driver)" ? Or something else ? 
no logic connection isn't precise enough